rskj-core/src/main/java/co/rsk/peg/whitelist/WhitelistSupportImpl.java

```diff
@@ -0,0 +1,226 @@
+        this.signatureCache = signatureCache;
+        this.btcContext = btcContext;
+    }
+
+    /**
+     * Returns the lock whitelist size, that is, the number of whitelisted addresses
+     *
+     * @return the lock whitelist size
+     */
+    @Override
+    public int getLockWhitelistSize() {
+        return whitelistStorageProvider.getLockWhitelist().getSize();
+    }
+
+    /**
+     * Returns the lock whitelist address stored at the given index, or null if the index is out of
+     * bounds
+     *
+     * @param index the index at which to get the address
+     * @return the base58-encoded address stored at the given index, or null if index is out of
+     * bounds
+     */
+    @Override
+    public LockWhitelistEntry getLockWhitelistEntryByIndex(int index) {
+        List<LockWhitelistEntry> entries = whitelistStorageProvider.getLockWhitelist().getAll();
+
+        if (index < 0 || index >= entries.size()) {
+            return null;
+        }
+        return entries.get(index);
+    }
+
+    @Override
+    public LockWhitelistEntry getLockWhitelistEntryByAddress(String addressBase58) {
+        try {
+            Address address = Address.fromBase58(btcContext.getParams(), addressBase58);
+
+            return whitelistStorageProvider.getLockWhitelist().get(address);
+        } catch (AddressFormatException e) {
+            logger.warn(INVALID_ADDRESS_FORMAT_MESSAGE, e);
+            return null;
+        }
+    }
+
+    /**
+     * Adds the given address to the lock whitelist.
+     *
+     * @param addressBase58    the base58-encoded address to add to the whitelist
+     * @param maxTransferValue the max amount of satoshis enabled to transfer for this address
+     * @return 1 SUCCESS, -1 ADDRESS_ALREADY_WHITELISTED, -2 INVALID_ADDRESS_FORMAT,
+     * -10 UNAUTHORIZED_CALLER, 0 UNKNOWN_ERROR.
+     */
+    @Override
+    public int addOneOffLockWhitelistAddress(Transaction tx, String addressBase58, BigInteger maxTransferValue) {
+        try {
+            Address address = Address.fromBase58(btcContext.getParams(), addressBase58);
+            Coin maxTransferValueCoin = Coin.valueOf(maxTransferValue.longValueExact());
+            LockWhitelistEntry entry = new OneOffWhiteListEntry(address, maxTransferValueCoin);
+
+            return this.addLockWhitelistAddress(tx, entry);
+        } catch (AddressFormatException e) {
+            logger.warn(INVALID_ADDRESS_FORMAT_MESSAGE, e);
+            return INVALID_ADDRESS_FORMAT.getCode();
+        }
+    }
+
+    @Override
+    public int addUnlimitedLockWhitelistAddress(Transaction tx, String addressBase58) {
+        try {
+            Address address = Address.fromBase58(btcContext.getParams(), addressBase58);
+            LockWhitelistEntry entry = new UnlimitedWhiteListEntry(address);
+
+            return this.addLockWhitelistAddress(tx, entry);
+        } catch (AddressFormatException e) {
+            logger.warn(INVALID_ADDRESS_FORMAT_MESSAGE, e);
```

Comment:
In some other classes and logs we use the format `logger.type("[methodName] message"); should we do the same from now on?

In this case, something like:

`ogger.warn("[addUnlimitedLockWhitelistAddress] {}", INVALID_ADDRESS_FORMAT_MESSAGE, e);`

What do you guys think? @wilmerrootstock , @julia-zack , @apancorb , @nathanieliov , @marcos-iov.

Reply:
Good catch

Reply:
Done! Good catch, @jeremy-then! It would be nice to have this as a standard and documented.